import "goby/common/protobuf/option_extensions.proto";
import "goby/common/protobuf/app_base_config.proto";
import "goby/acomms/protobuf/queue.proto";
import "goby/acomms/protobuf/route.proto";
import "goby/acomms/protobuf/amac_config.proto";
import "goby/acomms/protobuf/driver_base.proto";

package goby.acomms.protobuf;

message BridgeConfig
{
  optional AppBaseConfig base = 1;

  message Subnet
  {
    required QueueManagerConfig queue_cfg = 1;
    required MACConfig mac_cfg = 2;
    required DriverType driver_type = 3;
    required DriverConfig driver_cfg = 4;
  }
  
  repeated Subnet subnet = 2;  
  
  optional RouteManagerConfig route_cfg = 3;

  repeated string load_shared_library = 10 [(goby.field).description="Path to a shared library containing compiled DCCL protobuf files. Preferred over load_proto_file."];
  
  repeated string load_proto_file = 11 [(goby.field).description="Path to a DCCL protobuf file. Use load_shared_library when possible."];

  repeated string load_dccl_message = 12 [(goby.field).description="Full name of DCCL protobuf message to validate and load. Must be compiled in or previously loaded using either load_shared_library or load_proto_file"];
  
}
